lib/cisco_node_utils/snmpuser.rb

```diff
@@ -74,32 +74,39 @@ def initialize_validator(name, groups, authproto, authpass, privproto,
       end
     end
 
-    ENGINE_ID_PATTERN = /([0-9]{1,3}(:[0-9]{1,3}){4,31})/
     def self.users
       users_hash = {}
       # config_get returns hash if 1 user, array if multiple, nil if none
       users = config_get('snmp_user', 'user')
       return users_hash if users.nil?
-      users = [users] if users.is_a?(Hash)
       users.each do |user|
-        name = user['user']
-        engineid = user['engineID']
-        if engineid.nil?
+        # n7k has enforcepriv, use-ipv*acl, avoid them
+        next if user[/(enforcePriv|use-ipv4acl|use-ipv6acl)/]
```

Comment:
Putting platform logic in the yaml is preferable but sometimes it's unavoidable in the ruby code. My concern with this logic you've added here is that it seems like it could skip over legitimate user names  ('snmp-server user use-ipv4acl...') - perhaps the regexp you are defining in the YAML (`config_get_token: '/^snmp.server user (.*)$/'`) needs to be more narrowly defined?